includes/core-changes/serialization/5.0/deserializing-json-into-char-requires-single-character.md

```diff
@@ -0,0 +1,46 @@
+### JsonSerializer.Deserialize requires single-character string
+
+When the type parameter is <xref:System.Char>, the string argument to <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> must contain a single character for deserialization to succeed.
+
+#### Change description
+
+In previous .NET versions, if you pass a multi-character string to <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> and the type parameter is <xref:System.Char>, the deserialization succeeds and only the first character is deserialized.
+
+In .NET 5.0 and later, when the type parameter is <xref:System.Char>, passing anything other than a single-character string causes a <xref:System.Text.Json.JsonException> to be thrown.
+
+```csharp
+// .NET Core 3.0 and 3.1: Returns the first character 'a'.
+// .NET 5.0 and later: Throws JsonException because payload has more than one character.
+JsonSerializer.Deserialize<char>("abc");
+
+// Correct usage.
+JsonSerializer.Deserialize<char>("a");
```

Comment:
Both these are invalid JSON because they are missing the string within embedded quotes, These should be:
```suggestion
JsonSerializer.Deserialize<char>("\"abc\"");

// Correct usage.
JsonSerializer.Deserialize<char>("\"a\"");
```

Reply:
But then the first character is always ", so how do you pass a single-character string?

Reply:
That **is** how you pass a single character string.

Here's an example valid, single-character JSON string:
`"a"`

In JSON, a string token is one that is surrounded with quotes.

Another valid JSON token, is number, so for example:
`1`

The way you would write that in C# is as follows (note, there are no extra quotes):
```C#
JsonSerializer.Deserialize<int>("1");
```

Maybe if you think of it as a character array rather than strings, that might clear up the need for the quotes:
```C#
byte[] arr = {(byte)'"', (byte)'a', (byte)'"'};
Console.WriteLine(JsonSerializer.Deserialize<char>(arr.AsSpan())); // valid, character 'a'

byte[] another = {(byte)'1', (byte)'2', (byte)'3'};
Console.WriteLine(JsonSerializer.Deserialize<int>(another.AsSpan())); // valid, number 123
```

Reply:
You can explore various values in jsonlint to see:
https://jsonlint.com/

![image](https://user-images.githubusercontent.com/6527137/95281446-09160580-080c-11eb-9ce9-16dc3d42d6e3.png)

Reply:
> Both these are invalid JSON because they are missing the string within embedded quotes, These should be:

Ah, forgot about that